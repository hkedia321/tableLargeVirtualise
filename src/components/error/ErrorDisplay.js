import React from 'react';
import PropTypes from 'prop-types';

const ErrorDisplay = React.memo(function(props){
    return (
        <h4 data-testid="errorDisplay" align="center" style={{color: '#F44336'}}>
            Error: {props.message}
        </h4>
    )
})

ErrorDisplay.propTypes = {
    message: PropTypes.string
}

export default ErrorDisplay;