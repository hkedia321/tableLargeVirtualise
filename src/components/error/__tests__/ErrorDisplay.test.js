import React from 'react';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import ErrorDisplay from '../';

afterEach(cleanup);

it('renders errorDisplay without crashing', () => {
    const { getByTestId } = render(<ErrorDisplay/>);
    expect(getByTestId('errorDisplay'));
})

it('renders errorDisplay correctly', () => {
    const { getByTestId } = render(<ErrorDisplay message="error"/>);
    expect(getByTestId('errorDisplay')).toHaveTextContent("Error: error");
})

it('matches snapshot', () => {
    const tree = renderer.create(<ErrorDisplay message="error"/>).toJSON();
    expect(tree).toMatchSnapshot();
})