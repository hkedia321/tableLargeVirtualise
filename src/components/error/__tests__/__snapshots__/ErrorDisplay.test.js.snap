// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<h4
  align="center"
  data-testid="errorDisplay"
  style={
    Object {
      "color": "#F44336",
    }
  }
>
  Error: 
  error
</h4>
`;
